fcrepo-http-commons/src/main/java/org/fcrepo/http/commons/exceptionhandlers/ExceptionDebugLogging.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2015 DuraSpace, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.http.commons.exceptionhandlers;
+
+import org.slf4j.Logger;
+
+/**
+ * @author barmintor
+ * @since 6/28/16
+ */
+public interface ExceptionDebugLogging {
+    /**
+     * Log a Throwable at the DEBUG level, log the stacktrace at the TRACE level.
+     *
+     * @param context ExceptionDebugLogging the exception intercepting context.
+     * @param error Throwable the intercepted error.
+     * @param logger Logger the logger to use
+     */
+    public default void debugException(ExceptionDebugLogging context, Throwable error, Logger logger) {
```

Comment:
This may be more a matter of style, but my preference would be to factor out `logger` into an abstract method `Logger logger()` and let it be implemented at greatest common factors in the mapper type-hierarchy.